src/Payment.ts

```diff
@@ -3,159 +3,249 @@ import { Utils } from './Utils'
-          body: JSON.stringify(data)
-        })
-        .then(({ estimatedGas, fees, path }) => ({
-          estimatedGas,
-          path,
-          maxFees: utils.formatAmount(fees, dec)
-        }))
-        .catch(e => Promise.reject(e))
+    options: PaymentOptions = {}
+  ): Promise<PathObject> {
+    try {
+      const { _currencyNetwork, _utils, _user } = this
+      let { decimals, maximumHops, maximumFees} = options
+      decimals = await _currencyNetwork.getDecimals(network, decimals)
+      const data = {
+        from: accountA,
+        to: accountB,
+        value: this._utils.calcRaw(value, decimals)
+      }
+      if (maximumFees) {
+        data['maxFees'] = maximumFees
+      }
+      if (maximumHops) {
+        data['maxHops'] = maximumHops
+      }
+      const endpoint = `networks/${network}/path-info`
+      const { estimatedGas, fees, path } = await _utils.fetchUrl<PathRaw>(endpoint, {
+        method: 'POST',
+        headers: new Headers({ 'Content-Type': 'application/json' }),
+        body: JSON.stringify(data)
       })
+      return {
+        estimatedGas,
+        path,
+        maxFees: _utils.formatAmount(fees, decimals)
+      }
+    } catch (error) {
+      return Promise.reject(error)
+    }
   }
 
-  public get (networkAddress: string, filter?: object): Promise<any> {
-    const mergedFilter = Object.assign({ type: 'Transfer' }, filter)
-    return this.event.get(networkAddress, mergedFilter)
+  /**
+   * Returns transfer logs of loaded user in a specified currency network.
+   * @param network Address of currency network.
+   * @param filter Event filter object. See `EventFilterOptions` for more information.
+   */
+  public get (
+    network: string,
+    filter: EventFilterOptions = {}
+  ): Promise<TLEvent[]> {
+    return this._event.get(network, {
+      ...filter,
+      type: 'Transfer'
+    })
   }
 
-  public confirm (rawTx): Promise<string> {
-    return this.user.signTx(rawTx).then(signedTx => this.transaction.relayTx(signedTx))
+  /**
+   * Signs and relays a raw transaction as returned by `prepare`.
+   * @param rawTx RLP encoded hex string defining the transaction.
+   */
+  public async confirm (rawTx): Promise<string> {
+    try {
+      const signedTx = await this._user.signTx(rawTx)
+      return this._transaction.relayTx(signedTx)
+    } catch (error) {
+      return Promise.reject(error)
+    }
   }
 
-  public createRequest (network: string, amount: number, subject: string): Promise<string> {
-    return new Promise((resolve, reject) => {
-      const params = [ 'paymentrequest', network, this.user.address, amount, subject ]
-      resolve(this.utils.createLink(params))
+  /**
+   * Creates a payment request link.
+   * @param network Address of a currency network.
+   * @param amount Requested transfer amount.
+   * @param subject Additional information for payment request.
+   */
+  public createRequest (
+    network: string,
```

Comment:
networkAddress

Reply:
moved to https://github.com/trustlines-network/clientlib/issues/87